src/components/Filters/FiltersWithRedux.jsx

```diff
@@ -0,0 +1,214 @@
+import "./Filters.css";
+import { Gridd } from "../Grid/Grid";
+import { useEffect } from "react";
+import { useDispatch, getState } from "react-redux";
+
+
+import {
+  filterByCategory,
+  filterByRating,
+  getAllData,
+  filterLowtoHigh,
+  filterHightoLow,
+} from "../../redux/Slice";
+
+
+const Axios = require("axios");
+
+export const FilterWithRedux = () => {
+  const dispatch = useDispatch();
+
+  //Functions to get data acc. to clicked filter
+
+  useEffect(() => {
+    //Get Default Data
+    const showData = () => {
+      Axios.get("http://localhost:3001/data").then((res) => {
+        dispatch(getAllData(res.data));
+      });
+    };
+    showData();
+  }, [dispatch]);
+
+  
+  const showDataByCategory = (exp) => {
+    //Get Data By Category
+    Axios.get(`http://localhost:3001/data?category=${exp}`).then((res) => {
+      dispatch(filterByCategory(res.data));
+    });
+  };
+  
+
+  const showDataByRating = (exp) => {
+    //Get Data By Rating
+    Axios.get(`http://localhost:3001/data?rating_gte=${exp}`).then((res) => {
+      dispatch(filterByRating(res.data));
+    });
+  };
+
+  const showDataLowtoHigh = (exp) => {
+    //Get Data sort L to H
+    Axios.get(`http://localhost:3001/data?_sort=${exp}&_order=asc`).then(
+      (res) => {
+        dispatch(filterLowtoHigh(res.data));
+      }
+    );
+  };
+
+  const showDataHightoLow = (exp) => {
+    //Get Data sort H to L
+    Axios.get(`http://localhost:3001/data?_sort=${exp}&_order=desc`).then(
+      (res) => {
+        dispatch(filterHightoLow(res.data));
+      }
+    );
+  };
+
+  //Functions to trigger the right above mentioned functions on filter click
+
+  const filterClicked = (exp) => {
+    //To check which function to execute when filter is clicked
+    if (exp == 1 || exp == 2 || exp == 3 || exp == 4) {
```

Comment:
there should only be one function to make api calls. 

We should have a state variable `const [filter, setFilter] = useState({category: default category, sort: defualtSort, rating:default rating})`;

Then this state variable should be used to get values of filters at the time of making api calls.